### PR TITLE
Wordlink changes

### PR DESCRIPTION
**Pull request by spcheema**
### Summary of change
* Rename `UploadWordLinkRecording` endpoint to `UploadTextBackTranslation `
* Rename `UploadWordLinkRecording` integration test
* Remove `recording` reference from the code since it's no longer a recording
* Clean up UI and remove recoding playback file
* Update `UploadTextBackTranslation` endpoint to accept multiple translations i.e array of `textBackTranslation`
* Add `DeleteTextBackTranslation` endpoint & integration test
* Hide scroll from term list when term's list is searched
* Tweaked UI to underlined linked term in the slide section
* Moved wordlink tab and displayed as default

### CODE REVIEW

File: API/DeleteTextBackTranslation.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+$androidId = trim($_POST['PhoneId']);
+$term = trim($_POST['term']);
+
+$model = new Model();
+# check if project exists for the given androidId
+$projectId = $model->GetProjectId($androidId);
+
+if (!$projectId) {
+    RespondWithError(400, "Please register a project using /API/RegisterPhone.php before using /API/UploadTextBackTranslation.php");
```
Comment (johntseng) [inline API/DeleteTextBackTranslation.php]: please update to "API/DeleteTextBackTranslation.php"

File: API/DeleteTextBackTranslation.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+
+if ($model->DeleteTextBacktranslation($projectId, $term)) {
+    exit;
```
Comment (johntseng) [inline API/DeleteTextBackTranslation.php]: Does this return a 200 or 204? Either one is fine.

File: docs/new-schema.sql
Comment (johntseng) [inline docs/new-schema.sql:1]: Can you also rename the table to WordlinkTranslations?

**Review by johntseng — CHANGES_REQUESTED** [review]

File: API/DeleteTextBackTranslation.php (hunk 2 above)
Comment (spcheema) [reply API/DeleteTextBackTranslation.php]: 200. My assumption was that it's a post request. Let me know if you want to change.

File: API/DeleteTextBackTranslation.php (hunk 1 above)
Comment (spcheema) [reply API/DeleteTextBackTranslation.php]: Ahhh... it's copy paste. I think we should return 404 instead 400.

Here is the error message order:
* 405 if request method is other than `POST`
* 400 if request doesn't have both `PhoneId` & `term`
* 404 if no word link translation found corresponding to the requested `PhoneId` & `term`
* 400 if project not found - **I think instead we should return 404** @johntseng 


Let's tell end user to create a project and then upload translations and then delete it.

I am happy with either way.

File: docs/new-schema.sql
Comment (spcheema) [reply docs/new-schema.sql:1]: Yeah sure... I'll update the table name in endpoints and schema file.

File: API/DeleteTextBackTranslation.php (hunk 2 above)
Comment (spcheema) [reply API/DeleteTextBackTranslation.php]: Actually changed 204 which make more sense

File: docs/new-schema.sql
Comment (spcheema) [reply docs/new-schema.sql:1]: Rename table query if needed to update your local DB or server:

`RENAME TABLE StoryProducer.WordLinkRecordings TO StoryProducer.WordlinkTranslations;`

File: API/DeleteTextBackTranslation.php (hunk 1 above)
Comment (spcheema) [reply API/DeleteTextBackTranslation.php]: > 400 if project not found - I think instead we should return 404

API will return 404 if project not found. Please let me know if want to revert it back to 400.

File: API/DeleteWordLinkBackTranslation.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+$androidId = trim($_POST['PhoneId']);
+$term = trim($_POST['term']);
+
+$model = new Model();
+# check if project exists for the given androidId
+$projectId = $model->GetProjectId($androidId);
+
+if (!$projectId) {
+    RespondWithError(404, "");
```
Comment (johntseng) [inline API/DeleteWordLinkBackTranslation.php]: Can you use the same error message for missing project id's? 
Please register a project using /API/RegisterPhone.php before using /API/UploadWordLinkBackTranslation.php

**Review by johntseng — APPROVED** [review]

File: API/DeleteWordLinkBackTranslation.php (hunk 3 above)
Comment (spcheema) [reply API/DeleteWordLinkBackTranslation.php]: I can do that but it doesn't feel good to ask a user to upload a wordlink before deleting it.

If we put end user’s shoes, then I am trying to delete a wordlink but I have entered the incorrect project. So I got 404. I shouldn’t be suggested to upload a project.

Lemme know if you don't agree then I would go ahead and replicate the same error message.

Comment (johntseng) [reply API/DeleteWordLinkBackTranslation.php]: Great point. So maybe something like "Please double check the PhoneId. A project with $androidId was not found."

Comment (spcheema) [reply API/DeleteWordLinkBackTranslation.php]: @johntseng Error message updated.

**Review by johntseng — APPROVED** [review]

**Pull request merged**